### PR TITLE
[4.4] docs: improve description for Factories BC

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
See https://forum.codeigniter.com/showthread.php?tid=87335&pid=411906#pid411906

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/changelogs/v4.4.0.rst [hunk 1]
```diff
@@ -53,6 +53,15 @@ For example, when you call ``model(\Myth\Auth\Models\UserModel::class)`` or
       - returns ``Myth\Auth\Models\UserModel`` even if ``preferApp`` is true (default)
       - returns ``App\Models\UserModel`` if you define ``Factories::define('models', 'Myth\Auth\Models\UserModel', 'App\Models\UserModel')`` before calling the ``model()``
 
+If you had passed a non-existent classname by mistake, the previous version
+would have returned a class instance in the ``App`` or ``Config`` namespace
+because of the ``preferApp`` feature.
+
+For example, in a controller (``namespace App\Controllers``), if you call
```
Comment (MGatner) [inline user_guide_src/source/changelogs/v4.4.0.rst:60]:
```suggestion
For example, in a controller (``namespace App\Controllers``), if you called
```

File: user_guide_src/source/changelogs/v4.4.0.rst [hunk 2]
```diff
@@ -53,6 +53,15 @@ For example, when you call ``model(\Myth\Auth\Models\UserModel::class)`` or
...
+``config(Config\App::class)`` by mistake (note that you forgot the leading ``\``
```
Comment (MGatner) [inline user_guide_src/source/changelogs/v4.4.0.rst:61]:
```suggestion
``config(Config\App::class)`` by mistake (note the class is missing the leading ``\``),
```

File: user_guide_src/source/changelogs/v4.4.0.rst [hunk 3]
```diff
@@ -53,6 +53,15 @@ For example, when you call ``model(\Myth\Auth\Models\UserModel::class)`` or
...
+in ``Config\App::class``), that means you pass ``App\Controllers\Config\App``.
```
Comment (MGatner) [inline user_guide_src/source/changelogs/v4.4.0.rst:62]:
```suggestion
meaning you actually passed ``App\Controllers\Config\App``.
```

File: user_guide_src/source/changelogs/v4.4.0.rst [hunk 4]
```diff
@@ -53,6 +53,15 @@ For example, when you call ``model(\Myth\Auth\Models\UserModel::class)`` or
...
+But the class does not exist, so now Factories return ``null``.
```
Comment (MGatner) [inline user_guide_src/source/changelogs/v4.4.0.rst:63]:
```suggestion
But that class does not exist, so now Factories will return ``null``.
```

**Review by MGatner — CHANGES_REQUESTED** [review]
I like the change to "with namespace". Some readability suggestions.

**Comment by kenjis** [comment]
Updated!
Screenshot from 2023-08-21 10-02-37

**Review by MGatner — APPROVED** [review]

**Pull request merged**